currency/aggregate_conversions.go

```diff
@@ -0,0 +1,45 @@
+package currency
+
+import "github.com/prebid/prebid-server/errortypes"
+
+// AggregateConversions contains both the request-defined currency rate
+// map found in request.ext.prebid.currency and the currencies conversion
+// rates fetched with the RateConverter object defined in rate_converter.go
+// It implements the Conversions interface.
+type AggregateConversions struct {
+	customRates, serverRates Conversions
+}
+
+// NewAggregateConversions expects both customRates and pbsRates to not be nil
+func NewAggregateConversions(customRates, pbsRates Conversions) *AggregateConversions {
+	return &AggregateConversions{
+		customRates: customRates,
+		serverRates: pbsRates,
+	}
+}
+
+// GetRate returns the conversion rate between two currencies prioritizing
+// the customRates currency rate over that of the PBS currency rate service
+// returns an error if both Conversions objects return error.
+func (re *AggregateConversions) GetRate(from string, to string) (float64, error) {
+	rate, err := re.customRates.GetRate(from, to)
+
+	if err == nil {
+		if _, isMissingRateErr := err.(*errortypes.NoConversionRate); !isMissingRateErr {
```

Comment:
Wouldn't this line lead to a panic? In the previous line you're checking if `err` is `nil` and entering this block only if it is and here you're trying to see if you can cast `err` to `*errortypes.NoConversionRate`

I'd suggest changing this to be more along the lines of:
```
if err == nil {
	return rate, nil
} else if _, isMissingRateErr := err.(*errortypes.NoConversionRate); !isMissingRateErr {
	return nil, err
}
```

Reply:
Agree, done